audio/src/main/java/edu/artic/audio/AudioLookupFragment.kt

```diff
@@ -73,25 +74,31 @@ class AudioLookupFragment : BaseViewModelFragment<AudioLookupViewModel>() {
 
         registerNumPadSubscription()
 
-        getAudioServiceObservable()
+        getAudioServiceObservable(fm = childFragmentManager)
                 .bindTo(audioService)
                 .disposedBy(disposeBag)
 
         viewModel.lookupResults
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribeBy { result ->
                     when (result) {
-                        is LookupResult.FoundAudio -> sendToPlayerFragment(result)
+                        is LookupResult.FoundAudio -> playAndDisplay(result)
                         is LookupResult.NotFound -> shakeLookupField()
                     }
                 }
                 .disposedBy(disposeBag)
 
     }
 
-    private fun sendToPlayerFragment(foundAudio: LookupResult.FoundAudio) {
+    private fun playAndDisplay(foundAudio: LookupResult.FoundAudio) {
         audioService.subscribeBy {
             it?.playPlayer(foundAudio.hostObject)
+            val host = baseActivity
+            host.runOnUiThread {
```

Comment:
we should just do an `observeOn(AndroidSchedulers.main())` on `audioService` subject.

Reply:
Won't that make the `playPlayer` call execute on the main thread too?

Reply:
Resolved it since @Cliabhach updated the code.